public/components/ImageMosaic.vue

```diff
@@ -87,59 +87,47 @@ import {
   bulkRowSelectionUpdate,
 } from "../util/row";
 import { MULTIBAND_IMAGE_TYPE } from "../util/types";
-import { getImageFields } from "../util/data";
+import { getImageFields, sameData } from "../util/data";
+
+interface Field {
+  key: string;
+  type: string;
+}
 
 export default Vue.extend({
-  name: "image-mosaic",
+  name: "ImageMosaic",
 
   components: {
     ImageLabel,
     ImagePreview,
   },
 
   props: {
-    instanceName: String as () => string,
-    dataItems: Array as () => TableRow[],
-    dataFields: Object as () => Dictionary<TableColumn>,
-    labelFeatureName: { type: String, default: "" },
-    dataset: String as () => string,
+    instanceName: { type: String as () => string, default: "" },
+    dataItems: { type: Array as () => TableRow[], default: null },
+    dataFields: {
+      type: Object as () => Dictionary<TableColumn>,
+      default: () => {
+        return {};
+      },
+    },
+    labelFeatureName: { type: String as () => string, default: "" },
+    dataset: { type: String as () => string, default: "" },
   },
 
   data() {
     return {
       imageWidth: 128,
       imageHeight: 128,
       currentPage: 1,
+      previousPage: 0,
```

Comment:
Might be an unused variable.

Reply:
Yup, left over from an initial attempt at a fix.